arangod/Agency/Agent.h

```diff
@@ -508,6 +524,11 @@ class Agent final : public arangodb::Thread, public AgentInterface {
   // lock for _ongoingTrxs
   arangodb::Mutex _trxsLock;
 
+  // @brief promises for poll interface and the guard (mvm-style
+  std::mutex _promLock;
+  index_t _lowestPromise;
+  std::map<SteadyTimePoint, futures::Promise<query_t>> _promises;
```

Comment:
Please add a comment saying that the key is the expiration time. Maybe explain memory lifetime of the Promise and when it is deleted from this map.

Reply:
done.